model_utils/models.py

```diff
@@ -74,12 +72,14 @@ def save(self, *args, **kwargs):
         status_changed field is updated even if it is not given as
         a parameter to the update field argument.
         """
+        update_fields = kwargs.get('update_fields', None)
         if (
-            'update_fields' in kwargs
-            and 'status' in kwargs['update_fields']
-            and 'status_changed' not in kwargs['update_fields']
+            update_fields
+            and 'status' in update_fields
+            and 'status_changed' not in update_fields
         ):
-            kwargs['update_fields'] += ['status_changed']
+            kwargs['update_fields'] = list(kwargs['update_fields']) + ['status_changed']
```

Comment:
I think we should keep using `set` as above.

```suggestion
            kwargs['update_fields'] = set(update_fields).union({'status_changed'})
```

Reply:
```
        update_fields = kwargs.get('update_fields', None)
        if update_fields is not None:
            update_fields = set(update_fields)
            if (
                update_fields
                and 'status' in update_fields
            )
        ):
            kwargs['update_fields'] = update_fields.union({'status_changed'})
```

changing the code here to above would do the trick i suppose